src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -801,6 +801,9 @@ public void Disconnect(string reason, Exception exception = null)
 
             if (this.IsConnected) this.SetStateAsync(NetworkPeerState.Disconnecting).GetAwaiter().GetResult();
 
+            this.Behaviors?.ToList().ForEach(b => b?.Detach());
```

Comment:
+1 to @monsieurleberre for the courage, but -2 for the Linq usage and most importantly -99 for corrupting the code with this change.

Reply:
+110 to boost my damaged ego, that was not my idea, and it is getting reverted ;)